backend/cmm_helpers.ml

```diff
@@ -4362,7 +4385,7 @@ let store ~dbg kind init ~addr ~new_value =
 let direct_call ~dbg ty pos f_code_sym args =
   Cop (Capply (ty, pos), f_code_sym :: args, dbg)
 
-let indirect_call ~dbg ty pos alloc_mode f args_type args =
+let rec indirect_call ~dbg ty pos alloc_mode f args_type args =
```

Comment:
I think we can merge both functions now that we have the changes in `to_cmm`, this would avoid the `~recurse` argument in `might_split_call_caml_apply`.